src/core/agents.jl

```diff
@@ -414,67 +253,75 @@ function compute_base_fields(base_type_spec)
     return base_fields
 end
 
-"""
-    kindof(agent::AbstractAgent) → kind::Symbol
-
-Return the "kind" (instead of type) of the agent, which is the name given to the
-agent subtype when it was created with [`@multiagent`](@ref).
-"""
-function DynamicSumTypes.kindof(a::AbstractAgent)
-    throw(ArgumentError("Agent of type $(typeof(a)) has not been created via `@multiagent`."))
-end
-
-"""
-    allkinds(AgentType::Type) → kinds::Tuple
-
-Return all "kinds" that compose the given agent type that was generated via
-the [`@multiagent`](@ref) macro. The kinds are returned as a tuple of `Symbol`s.
-"""
-function DynamicSumTypes.allkinds(a::Type{<:AbstractAgent})
-    (nameof(a), ) # this function is extended automatically in the macro
-end
+###########################################################################################
+# @multiagent
+###########################################################################################
 
 """
-    @dispatch f(args...)
+    @multiagent YourAgentType(AgentTypesToMerge) [<: OptionalSupertype]
 
-A macro to enable multiple-dispatch-like behavior for the function `f`,
-for various agent kinds generated via the [`@multiagent`](@ref) macro.
-For an illustration of its usage see the [Tutorial](@ref).
+Define multiple agent "subtypes", which are variants of a unique type `YourAgentType`. 
+This means that all "subtypes" are enclosed in the overarching type. Then, You cannot
+distinguish them on the basis of `typeof`, but need to use instead the `variantof`
+function. The `allvariants` function for a convenient way to obtain all variants types.
```

Comment:
Can you please move the reference to DynamicSumTypes.jl here and say that it uses the `@sumtype` macro, instead of referencing DynamicSumTypes.jl in the main tutorial?